NMS-7712: Migrate documentation from opennms.properties

indigo423 opened the pull request:
Migrated all documentation from opennms.properties to Admin guide. The described default values are set in the opennms.properties. For the reason the properties are used in several parts of OpenNMS they are ordered in sections which are reflected also in the documentation part.

JIRA: http://issues.opennms.org/browse/NMS-7712

soleger commented:
* Many of the options that were commented out are now uncommented. I would leave all options that were previously commented out as commented out in the new version to make the new config functionally identical.
* I would prefer to follow the pattern used by PostgreSQL where all options in the config are commented out and represent the default values. This may require code changes to ensure that default values are used for certain system properties.
* I would like to see all possible values documented inside the config file itself.
* I would like to see a link to the documentation inside the config file itself.

I'm making all of the recommendations at once since any change to opennms.properties requires manual merging by our support staff for every customer and I'd rather cluster all drastic changes to the file so that they all get done at once.

soleger commented on opennms-base-assembly/src/main/filtered/etc/opennms.properties:
```diff
@@ -1,622 +1,126 @@
+opennms.bin.dir=${install.bin.dir}
+java.awt.headless=true
+org.opennms.dao.ipinterface.findByServiceType=select distinct ipIf from OnmsIpInterface as ipIf join ipIf.monitoredServices as monSvc where monSvc.serviceType.name = ?
+org.opennms.web.aclsEnabled=false
```
This should be moved to "Section :: WEB UI".

pbrane commented:
+1

> On Jun 5, 2015, at 17:15, Seth Leger <notifications@github.com> wrote:
> 
> Many of the options that were commented out are now uncommented. I would leave all options that were previously commented out as commented out in the new version to make the new config functionally identical.
> I would prefer to follow the pattern used by PostgreSQL where all options in the config are commented out and represent the default values. This may require code changes to ensure that default values are used for certain system properties.
> I would like to see all possible values documented inside the config file itself.
> I would like to see a link to the documentation inside the config file itself.
> I'm making all of the recommendations at once since any change to opennms.properties requires manual merging by our support staff for every customer and I'd rather cluster all drastic changes to the file so that they all get done at once.
> 
> —
> Reply to this email directly or view it on GitHub.
>

indigo423 commented:
@soleger I've worked in your suggestions regarding uncommented and commented settings and moved some parts around. I've dropped a mail to opennms-discuss and opennms-docs to get some feedback about this issue and tried to explain the motivations behind this change. Thanks for your feedback, I'll dig in the PostgreSQL pattern to get more familiar with it.

soleger commented on opennms-base-assembly/src/main/filtered/etc/opennms.properties:
```diff
@@ -1,622 +1,179 @@
-# which is required by Rtcd to post realtime status updates.  If you do not
-# wish to allow unsecured HTTP access to the OpenNMS web UI, you must set
-# org.opennms.netmgt.jetty.host above or use firewall rules to accomplish this.
 #org.opennms.netmgt.jetty.https-port = 8443
```
The Jetty HTTPS stuff needs to be in its own section apart from normal Jetty properties. Also, some of this might be obsolete since we now use jetty.xml to run the server. Only the properties that are referenced in the default jetty.xml file should be necessary now.

indigo423 replied:
Done

soleger commented on opennms-base-assembly/src/main/filtered/etc/opennms.properties:
```diff
@@ -1,622 +1,179 @@
-# You will need to adjust alarmTroubleTicketLinkTemplate to suit your OTRS Base URL
-# Note that you will need to install the OpenNMS module for OTRS in your OTRS server
-
 #opennms.ticketer.plugin=org.opennms.netmgt.ticketer.otrs.OtrsTicketerPlugin
```
This section needs a one-line explanation that all of these properties are used to enable OTRS ticketing.

soleger commented on opennms-base-assembly/src/main/filtered/etc/opennms.properties:
```diff
@@ -1,622 +1,179 @@
-# to enable RtTicketerPlugin
-# You will need to adjust alarmTroubleTicketLinkTemplate to suit your RT Base URL
-
 #opennms.ticketer.plugin=org.opennms.netmgt.ticketer.rt.RtTicketerPlugin
```
This section needs a one-line explanation that all of these properties are used to enable RT ticketing.

soleger commented on opennms-base-assembly/src/main/filtered/etc/opennms.properties:
```diff
@@ -1,622 +1,179 @@
-# You will need to adjust alarmTroubleTicketLinkTemplate to suit your Remedy Base URL
-# The following settings are for locale it_IT
-
 #opennms.ticketer.plugin=org.opennms.netmgt.ticketer.remedy.RemedyTicketerPlugin
```
This section needs a one-line explanation that all of these properties are used to enable Remedy ticketing.

soleger commented on opennms-base-assembly/src/main/filtered/etc/opennms.properties:
```diff
@@ -1,622 +1,179 @@
-###### EVENTD IPC ######
-# The hostname or IP address of the OpenNMS server where events should be sent.
-# Default: 127.0.0.1
 #opennms.rtc.event.proxy.host=127.0.0.1
```
These should all be in the RTC IPC section.

soleger commented on opennms-base-assembly/src/main/filtered/etc/opennms.properties:
```diff
@@ -1,622 +1,179 @@
-# (OpenStreetMaps API)
+gwt.maptype=OpenLayers
+gwt.apikey=
+#gwt.geocoder.class=org.opennms.features.poller.remote.gwt.server.geocoding.GoogleMapsGeocoder
```
The geocoder section needs an explanation that you are choosing one implementation among the 3 options (Google, Nominatim, Mapquest).

soleger commented on opennms-base-assembly/src/main/filtered/etc/opennms.properties:
```diff
@@ -1,622 +1,179 @@
+
+### General
+
+#org.opennms.dao.ipinterface.findByServiceType=select distinct ipIf from OnmsIpInterface as ipIf join ipIf.monitoredServices as monSvc where monSvc.serviceType.name = ?
```
This property needs to be DELETED!!! Why is there HQL in opennms.properties? This must be for some custom behavior somewhere... but whoever needs this should fork their own repo and make the change there.

soleger commented on opennms-doc/guide-admin/src/asciidoc/text/config-files/opennms.properties.adoc:
```diff
@@ -0,0 +1,920 @@
+These properties will be set as system properties at the beginning of Java startup.
+Any properties set on the Java command-line with `-Dproperty=value` will be overridden by these values.
+
+[options="autowidth"]
```
I'm not sure I like autowidth here because some of the panels are not as wide, making it difficult to scan down the document and see the property names easily. I would prefer width=100%.

indigo423 replied:
This is right now a more complicated topic and should be moved to a different topic for the reason it implies all tables in the doc we use right now and has implication on the PDF rendered output.

soleger commented on opennms-doc/guide-admin/src/asciidoc/text/config-files/opennms.properties.adoc:
```diff
@@ -0,0 +1,920 @@
+
+[options="header, autowidth"]
+|===
+| Property | `opennms.bin.dir`
```
Is it also possible to left-justify the property names so that they all line up vertically on the page? I don't like the centering since it makes it difficult to scan for property names.

soleger replied:
Also, is the 'Property' column necessary? People should know just from looking at the doc that this is a system property that is contained in opennms.properties. I think it just takes up space.

indigo423 replied:
The content is left-justified with rendering in our doc, the GitHubs preview is just a very simple way to represent the adoc file and tables. It's really just a preview. If you want to see the output generated with bamboo you can run a `mvn clean package` in the `guide-admin` directory.

indigo423 replied:
I have removed the 'Property' column, I agree about the space.

indigo423 commented on opennms-base-assembly/src/main/filtered/etc/opennms.properties:
```diff
@@ -1,622 +1,179 @@
+### General
+
+#org.opennms.dao.ipinterface.findByServiceType=select distinct ipIf from OnmsIpInterface as ipIf join ipIf.monitoredServices as monSvc where monSvc.serviceType.name = ?
+#org.opennms.netmgt.collectd.DefaultCollectionAgent.loadSnmpDataOnInit=false
```
Regarding the existing documentation this property is related to `org.opennms.dao.ipinterface.findByServiceType`. Should this property also be deleted?
I can deal with the deletion from the docs, I can't deal with the "...but whoever needs this should fork their own repo and make the change there" :)

soleger replied:
I meant: this property needs to be deleted and we need to remove usage of the property from the source code. I don't know its exact origin but it's not acceptable to have the ability to edit HQL statements by setting a system property.

indigo423 replied:
@soleger Is this something we can solve in this pull request? The topic is migrating the existing documentation. I would recommend to open an dedicated issue to remove this property. The only thing I can do is to remove the property from the doc. This is from my point of view the last open topic in this pull request. Thanks for taking care, helps a lot.

StCyr commented on opennms-doc/guide-admin/src/asciidoc/text/config-files/_template.adoc:
```diff
@@ -0,0 +1,23 @@
+|===
+| Type             | e.g. Java property file with key value pairs or XML file with XSD
+| Location         | `path where the configuration file is located`
+| Restart required | yes or no
```
>"restart required" 

It's not just yes or no, you can also have "send-event.pl"

StCyr commented on opennms-doc/guide-admin/src/asciidoc/text/config-files/_template.adoc:
```diff
@@ -0,0 +1,23 @@
+
+// REMOVE ME!! Please keep first line an empty line to make sure, the ToC can be build correctly
+==== File: {name-of-the-configuration-file}
```
Also, for xml files, there's already a lot of documentation in the schema files (.xsd files).

Couldn't you setup some automation to get some info from these files right here?

mvrueden commented:
Hey Ulf, I do not see an OCA check here, can you please perform that one for me, please?

indigo423 commented:
Undecided how to proceed, we keep the documentation in the current opennms.properties file.

j-white commented:
Was there any specific reason why we didn't proceed with this, or just a general of consensus?

indigo423 commented:
@j-white To sum it up very briefly, I did a proposal to move properties to official versioned documentation and remove the docs from the properties files to not have to maintain in two places. The proposal was rejected with the result to keep the documentation in the properties files so an administrator has the bits and pieces to configure properties on hand when he is editing the file.

indigo423 commented:
See also here: https://issues.opennms.org/browse/HZN-356
